tools/build-release-branch.sh

```diff
@@ -0,0 +1,208 @@
+if [[ -n $( git status -s --porcelain ) ]]; then
+ 	echo "Uncommitted changes found."
+ 	echo "Please deal with them and try again clean."
+#    exit 1
+fi
+
+# Check the command
+if [[ 'new' == $COMMAND || '-n' == $COMMAND ]]
+then
+    create_new_release_branches
+elif [[ 'update' = $COMMAND || '-u' = $COMMAND ]]
+then
+    # It's possible they passed the branch name directly to the script
+    if [[ -z $2 ]]
+    then
+        read -p "What branch are you updating? (enter full branch name): " branch
+        UPDATE_BUILT_BRANCH=$branch
+    else
+        UPDATE_BUILT_BRANCH=$2
+    fi
+else
+    usage
+fi
+
+
+# Cast the branch name that we'll be building to a single var.
+if [[ -n $NEW_BUILT_BRANCH ]]
+then
+    BUILD_TARGET=$NEW_BUILT_BRANCH
+elif [[ -n $UPDATE_BUILT_BRANCH ]]
+then
+    BUILD_TARGET=$UPDATE_BUILT_BRANCH
+else
+    echo ""
+    echo "No target branch specified.  How did you make it this far?"
+    exit 1
+fi
+
+
+
+### This bit is the engine that will build a branch and push to another one ####
+
+# Make sure we're trying to deploy something that exists.
+if [[ -z $( git branch -r | grep "$BUILD_TARGET" ) ]]; then
+    echo "Branch $BUILD_TARGET not found in git repository."
+    echo ""
+    exit 1
+fi
+
+read -p "You are about to deploy a new production build to the $BUILD_TARGET branch from the $CURRENT_BRANCH branch. Are you sure? [y/N]" -n 1 -r
+if [[ $REPLY != "y" && $REPLY != "Y" ]]
+then
+    exit 1
+fi
+echo ""
+
+echo "Building Jetpack"
+
+# Checking for yarn
+hash yarn 2>/dev/null || {
+    echo >&2 "This script requires you to have yarn package manager installed."
+    echo >&2 "Please install it following the instructions on https://yarnpkg.com. Aborting.";
+    exit 1;
+}
+
+yarn run distclean
+yarn cache clean
+yarn
+NODE_ENV=production yarn run build-client
+echo "Done"
+
+# Prep a home to drop our new files in. Just make it in /tmp so we can start fresh each time.
+rm -rf TMP_REMOTE_BUILT_VERSION
+rm -rf TMP_LOCAL_BUILT_VERSION
+
+echo "Rsync'ing everything over from Git except for .git and npm stuffs."
+rsync -r --exclude='*.git*' --exclude=node_modules $DIR/* TMP_LOCAL_BUILT_VERSION
+echo "Done!"
+
+echo "Purging paths included in .svnignore"
+# check .svnignore
+for file in $( cat "$DIR/.svnignore" 2>/dev/null ); do
+	# We want to commit changes to to-test.md as well as the testing tips.
+	if [ $file == "to-test.md" || $file == "docs/testing/testing-tips.md" ]; then
+		continue;
+	fi
+	rm -rf TMP_LOCAL_BUILT_VERSION/$file
+done
+echo "Done!"
+
+echo "Pulling latest from $BUILD_TARGET branch"
+git clone --depth 1 -b $BUILD_TARGET --single-branch git@github.com:Automattic/jetpack.git TMP_REMOTE_BUILT_VERSION
+echo "Done!"
+
+echo "Rsync'ing everything over remote version"
+rsync -r --delete TMP_LOCAL_BUILT_VERSION/* TMP_REMOTE_BUILT_VERSION
+echo "Done!"
+
+cd TMP_REMOTE_BUILT_VERSION
+
+echo "Finally, Committing and Pushing"
+git add .
+git commit -am 'New build'
```

Comment:
I may be wrong, but I have a feeling that we're not committing stuff that has changed in folders that are in `.gitignore`. I think that we need another step here: removing all entries from `.gitignore` that are related to built files.